crates/holochain_integrity_types/src/chain.rs

```diff
@@ -16,87 +16,96 @@ mod test;
 /// Filter source chain items.
 /// Starting from some chain position given as an [`ActionHash`]
 /// the chain is walked backwards to genesis.
-/// The filter can stop early by specifying the number of
-/// chain items to take and / or an [`ActionHash`] to consume until.
+/// The filter can stop early by specifying stop conditions:
+/// A maximum number of items is reached, a given [`ActionHash`] is found, or
+/// a given timestamp has been passed.
+/// Multiple `[`ActionHash`] can be provided. The filter will stop at and take the first one found.
+/// When providing a Timestamp, the filter will stop at and take the last item that has a
+/// Timestamp younger than the provided one. In the edge case of multiple actions having the same
```

Comment:
```suggestion
/// Timestamp older than the provided one. In the edge case of multiple actions having the same
```

Or maybe it's "newer", I've not got to the database query logic yet.